From eec38c06a5b90bbd539cda2378c8e8c9efe7e248 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jeremy=20Lain=C3=A9?= <jeremy.laine@m4x.org>
Date: Wed, 16 Mar 2022 11:15:42 +0100
Subject: [PATCH] [streams] initialise stream avg_frame_rate (fixes: #876)

We need to initialise an output video stream's average frame rate from
the codec context's framerate, otherwise the duration and FPS calculations
are wrong when writing video with FFmpeg >= 4.4.
---
 av/container/output.pyx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/av/container/output.pyx b/av/container/output.pyx
index c1a47b5b..9910dc93 100644
--- a/av/container/output.pyx
+++ b/av/container/output.pyx
@@ -107,6 +107,7 @@ cdef class OutputContainer(Container):
             codec_context.framerate.num = rate.numerator
             codec_context.framerate.den = rate.denominator
 
+            stream.avg_frame_rate = codec_context.framerate
             stream.time_base = codec_context.time_base
 
         # Some sane audio defaults
